src/main/java/com/melon/portfoliomanager/dtos/TransactionDto.java

```diff
@@ -0,0 +1,66 @@
+package com.melon.portfoliomanager.dtos;
+
+import jakarta.validation.constraints.DecimalMin;
+import jakarta.validation.constraints.Digits;
+import jakarta.validation.constraints.NotNull;
+
+public class TransactionDto {
+
+    @NotNull(message = "Invalid user input. The username is null.")
+    private String username;
+
+    @NotNull(message = "Invalid user input. The asset symbol is null.")
+    private String assetSymbol;
+
+    @NotNull
+    @DecimalMin("0.01")
+    @Digits(integer = 1000000000, fraction = 4, message = "Value must contain up to 4 digits in the fraction part and must be a positive number.")
+    private Double quantity;
```

Comment:
Why do you use the wrapper ```Double``` type instead of the primitive ```double``` type here ?
Same question for the ```price``` field as well.